Update the release documentation

anfredette opened the pull request:
I've updated this documentation based on notes from the past two releases.  We should try to follow the steps described in this document for the next release and take another pass at updating it if necessary.

msherif1234 commented on RELEASE.md:
```diff
@@ -64,177 +72,303 @@ A release for the bpfman project is comprised of the following major components:
+    - quay.io/bpfman-userspace
+- The new version appears at [crates.io](https://crates.io/crates/bpfman)
+- A new RPM is built and pushed to the [bpfman COPR
+  repository](https://copr.fedorainfracloud.org/coprs/g/ebpf-sig/bpfman/).
```
we also we need to undo the changes in bpfman/src/lib.rs and pointed back to latest right ? or revert the PR

msherif1234 commented on RELEASE.md:
```diff
@@ -64,177 +72,303 @@ A release for the bpfman project is comprised of the following major components:
-The following steps must be done by one of the [bpfman maintainers][bpfman-team],
-please always try and cut a release candidate before pushing an official major
-or minor release.
+- Run `make build-release-yamls` and add the yaml files generated to the release
```
pls make it clear u will run the above make command from `examples` directory

anfredette replied:
Done.

msherif1234 reviewed (commented):
did u intentionally remove the update to the community operator ?

anfredette commented:
> did u intentionally remove the update to the community operator ?

I think it's in there.  See line 261 of new doc.  If not, let me know what to add, or we can add the instructions in an update after we do the next release.

> also we have now split repo so should we split the release doc or create common docs repo under and have that doc there as we as any general doc ?

Good point.  We can discuss, but maybe we add a RELEASE.md file to bpfman-operator that says something like, we synchronize the releases.  Do bpfman first.  And, point to the instructions in bpfman?

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
+stability of a resource. For example, if a resource has not yet graduated to
+beta, it is still possible that it could either be removed from the API or
+changed in backward incompatible ways. For more information on API versions,
+refer to the [full Kubernetes API versioning
```
```suggestion
refer to the full [Kubernetes API versioning
```
I would remove "full" from the link

anfredette replied:
Done.

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
-        - `version = "x.x.x"`
-        - `bpfman = { version = "x.x.x", path = "./bpfman" }"`
-        - `bpfman-api = { version = "x.x.x", path = "./bpfman-api" }`
+To simplify the generation of the release notes details, we recommend using the
```
```suggestion
To simplify the generation of the release notes details, we are using the
```

anfredette replied:
Done.

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
-- Tag the release using the commit on `main` where the changelog update merged.
-  This can  be done using the `git` CLI or Github's [release][release]
+    - Run `cargo generate-lockfile`
+    - Add a new `examples` config directory for the release version.
```
```suggestion
    - Add a new `bpfman/examples/config/v0.x.y/` and `bpfman/examples/config/v0.x.y-selinux/` config directory for the release version by copying the latest release directory and running a search for the current release and replace with the new release.```

anfredette replied:
Done.

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
+
+- Create a new branch in your `bpfman` fork named, for example
+  `<githubuser>/release-x.y.z`, and use the new branch in the upcoming steps.
+- Make the following changes
```
Question:
Do you think it is worth adding a step to search though the code for the current version, `0.5.1` (without the `v`) and replace with the current release. There is documentation that references releases like this file, loading from RPM, etc, that may be nice to always point to latest release in the instructions? There are a few spots pointing to 0.4.2 currently, so if we add this step, we should do a one time scrub for spots with older releases too.

anfredette replied:
That makes sense.

anfredette replied:
Actually, this is a little tricky.  There are a lot of places where the old release needs to remain, and there are other dependencies that just happen to have the same version number as the current or older bpfman releases.

Billy99 replied:
So the one time scrub might a bit tricky maybe. Is there anywhere that needs to stay at 0.5.1? But then going forward, just change current release to next release everywhere

anfredette replied:
I added a bullet, but whoever is using it will need to use their judgment.  You may find the string in the following places that shouldn't be changed: changelogs, cargo.toml, cargo.lock, Containerfile.bpfman.openshift (which has a different version), example config files (at least)

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
+
+#### bpfman-operator Release
+
+- Create a new branch in your `bpfman` fork named, for example
```
```suggestion
- Create a new branch in your `bpfman-operator` fork, for example
```

anfredette replied:
Done.

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
+
+#### bpfman Release
+
+- Create a new branch in your `bpfman` fork named, for example
```
```suggestion
- Create a new branch in your `bpfman` fork, for example
```

anfredette replied:
Done.

Billy99 commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
-  Github's [release][release] page.
-- Tag the release using the commit on `main` where the changelog update merged.
-  This can  be done using the `git` CLI or Github's [release][release]
+    - Run `cargo generate-lockfile`
```
- Update the bpfman version in the [examples/Makefile][examples-makefile]:
    - `VERSION ?= x.y.z`

And add link at bottom of file.

anfredette replied:
Done.

anfredette commented on RELEASE.md:
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
-Each new release of bpfman is defined with a "bundle version" that
-represents the Git tag of a release, such as `v0.4.2`. This contains the
-components described above
+MAJOR: Incremented for incompatible API changes.
```
Done

anfredette commented:
Thanks for your reviews @msherif1234  and @Billy99!  I think I've addressed all of your comments.  Let me know if I missed anything.
